includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -1637,6 +1644,8 @@ public function process_upe_redirect_payment( $order_id, $intent_id, $save_payme
 
 			exit;
 		}
+
+		$this->unlock_order_payment( $order );
```

Comment:
```suggestion
		finally {
            $this->unlock_order_payment( $order );
        }

```

nit: I think wrapping this up with a `finally` would be safer as it guarantees execution of this line at the end, no matter what happens inside `process_order_for_confirmed_intent`.